### DISCUSSION TITLE
generateStaticParams with param that's not included in route

### DISCUSSION THREAD

**Discussion by SamKomesarook** [vercel/next.js#51430] [App Router]
Is there a way to pass a param from generateStaticParams to the generated page that isn't included in the route?

For example, if we want the generated routes to be easier to read, such as `blogs/get-started` or `blogs/how-it-works`, but we want an ID to easily query for the page content, is there a way to get the ID in generateStaticParams and pass it to the page without adding it to the URL?

**Comment by simpel** [comment]
Could you make a API endpoint that fetches all ID:s maybe and use that to generate the pages?

**Reply by simpel** [reply]
```typescript
export async function generateStaticParams() {
	const ids = await getIDs()

	const slugs: unknown[] = []

	for (const id of ids) {
		//some logic
	}

	return slugs
}
```

**Reply by SamKomesarook** [reply]
Thank you for the suggestion, but the id's are used to request the page data, which obviously needs to be accessable outside the generateStaticParams function...

**Reply by levipadre** [reply]
I'm looking for a similar solution. Have you found anything @SamKomesarook?
I'm also working with slugs, my structure is: app/[[...slug]]/page.ts
but I want to fetch page data by ID

**Comment by AgrYpn1a** [comment]
Any updated regarding this question?
